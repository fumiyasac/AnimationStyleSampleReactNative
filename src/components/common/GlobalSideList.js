/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  View
} from 'react-native';

import {
  Text,
  Separator
} from 'native-base';

export default class GlobalSideList extends React.Component {
  render() {
    const { headerTitle, listItems } = this.props;
    return (
      <View>
        <Separator bordered>
          <Text>{headerTitle}</Text>
        </Separator>
        {listItems}
      </View>
    );
  };
}
