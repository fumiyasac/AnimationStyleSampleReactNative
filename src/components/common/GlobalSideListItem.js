/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet
} from 'react-native';

import {
  Text,
  ListItem
} from 'native-base';

export default class GlobalSideListItem extends React.Component {
  render() {
    const { title, onPressListItem } = this.props;
    return (
      <ListItem style={styles.listItem} onPress={onPressListItem}>
        <Text style={styles.listItemText}>{title}</Text>
      </ListItem>
    );
  };
}

const styles = StyleSheet.create({
  listItem: {
    height: 50
  },
  listItemText: {
    fontSize: 12,
    fontWeight: 'bold',
    color: '#555555'
  }
});
