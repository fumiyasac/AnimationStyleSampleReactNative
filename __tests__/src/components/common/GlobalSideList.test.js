import 'react-native';
import React from 'react';
import GlobalSideList from '../../../../src/components/common/GlobalSideList';
import GlobalSideListItem from '../../../../src/components/common/GlobalSideListItem';

import renderer from 'react-test-renderer';

it('GlobalSideListコンポーネントが正しく表示されること', () => {

  let stubObject = {
    "headerTitle": "メニュー",
    "items": [
      { title: "フィード", selectedIndex: 0 },
      { title: "勉強日記", selectedIndex: 1 },
      { title: "達成記録", selectedIndex: 2 },
      { title: "お買い物", selectedIndex: 3 },
      { title: "お問い合わせ", selectedIndex: 4 }
    ]
  };

  const onPressListItemMock = jest.fn();

  const makeLists = () => { 
    return stubObject.items.map( (item, index) => {
      return (
        <GlobalSideListItem 
          key={index}
          title={item.title}
          onPressListItem={onPressListItemMock()} />
      );
    });
  };

  const tree = renderer.create(
    <GlobalSideList 
      headerTitle={stubObject.headerTitle}
      listItems={makeLists()} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
  