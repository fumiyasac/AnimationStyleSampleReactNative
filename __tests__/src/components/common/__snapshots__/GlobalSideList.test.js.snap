// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSideListコンポーネントが正しく表示されること 1`] = `
<View>
  <View
    bordered={true}
    style={
      Object {
        "backgroundColor": "#F0EFF5",
        "borderBottomWidth": 0.5,
        "borderColor": "#c9c9c9",
        "borderTopWidth": 0.5,
        "flex": 1,
        "height": 35,
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingTop": 12,
      }
    }
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#777",
          "fontFamily": "System",
          "fontSize": 12,
        }
      }
      uppercase={false}
      virtual={undefined}
    >
      メニュー
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          },
          Object {
            "height": 50,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#555555",
              "fontSize": 12,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        フィード
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          },
          Object {
            "height": 50,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#555555",
              "fontSize": 12,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        勉強日記
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          },
          Object {
            "height": 50,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#555555",
              "fontSize": 12,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        達成記録
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          },
          Object {
            "height": 50,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#555555",
              "fontSize": 12,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        お買い物
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      onPress={undefined}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          },
          Object {
            "height": 50,
          },
        ]
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#555555",
              "fontSize": 12,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        お問い合わせ
      </Text>
    </View>
  </View>
</View>
`;
